R/financial_functions.R

```diff
@@ -80,7 +80,7 @@ calculate_net_profits_shock_declining_technologies_carbon_tax <- function(data,
   data_over_shoot_increasing <- data %>% dplyr::filter(.data$overshoot_direction == "Increasing") %>%
     dplyr::mutate(
       production_compensation = .data$late_sudden - .data$baseline,
-      carbon_tax = ifelse(year > shock_year, .data$carbon_tax, 0),
+      carbon_tax = ifelse(year < shock_year, 0, .data$carbon_tax),
```

Comment:
shouldn't it be `<=` ?

Reply:
its the same in result - i prefer these kind of clear statements

Reply:
it's not the same, with `>` the carbon tax is applied 1 year after the shock year, while with `<` the carbon tax is applied on the shock year included. i.e. 1 year of carbon tax difference.

But then maybe it's better to apply it on the shock year included, meaning your modification corrects a bug ?

Reply:
i know thats why i change it shockyear is when carbon tax is applied ...

Reply:
like before 2030 shock year would have meant 2031 has the carbon tax, but we want it to be already applied in 2030